src/interface/state_descriptor.hpp

```diff
@@ -188,6 +183,10 @@ class StateDescriptor {
     if (FillDerivedMesh != nullptr) FillDerivedMesh(rc);
   }
 
+  void OutputDiagnostics(SimTime const &simtime, MeshData<Real> *rc) const {
+    if (OutputDiagnosticsMesh != nullptr) OutputDiagnosticsMesh(simtime, rc);
+  }
+
```

Comment:
I like the addition of `OutputDiagnostics` at the package level. But I notice it's not actually called anywhere in the driver/mesh. Maybe we could add a loop over packages at the driver level (either before or after loop) to spit out the diagnostics, so that the application doesn't have to write that bit of boiler plate?

Reply:
Thanks for the review @Yurlungur.   I suppose this boils down to whether or not we want to "impose" this to be some sort of default behavior.  I had expected that the downstream use would chose if/how/when to output any package diagnostics.  I'm not opposed to some default call to all packages' `OutputDiagnostics` somewhere, but I'm not entirely sure where it should go.  Can you offer guidance?

Reply:
Now has default implementation to cycle the packages and all the `Pre`/`PostStepDiagnostics`.